api/prometheus/v1/api.go

```diff
@@ -640,6 +659,23 @@ func (h *httpAPI) Flags(ctx context.Context) (FlagsResult, error) {
 	return res, json.Unmarshal(body, &res)
 }
 
+func (h *httpAPI) Runtimeinfo(ctx context.Context) (RuntimeinfoResult, error) {
+	u := h.client.URL(epRuntimeinfo, nil)
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return RuntimeinfoResult{}, err
+	}
+
+	_, body, _, err := h.client.Do(ctx, req)
```

Comment:
It looks like matches the other methods, but why we discard the status code completely? No point in marshaling if we don't check 2** status. :thinking:

Reply:
Like you mentioned this just uses what the other functions use, but the custom `.Do` does check for status :) https://github.com/prometheus/client_golang/blob/master/api/prometheus/v1/api.go#L880

Reply:
I think this is something to fix in follow up PRs if any, just it feels wrong, will add issue to follow up (:

Reply:
Agreed, we should fix it for all the functions!